pkg/ipoib/ipoib.go

```diff
@@ -90,11 +90,24 @@ func NewIpoibManager() types.Manager {
 func (im *ipoibManager) CreateIpoibLink(conf *types.NetConf, ifName string, netns ns.NetNS) (
 	*current.Interface, error) {
 	iface := &current.Interface{}
-	m, err := im.nLink.LinkByName(conf.Master)
+	lnk, err := im.nLink.LinkByName(conf.Master)
 	if err != nil {
 		return nil, fmt.Errorf("failed to lookup master %q: %v", conf.Master, err)
 	}
 
+	if lnk.Type() != "ipoib" {
+		return nil, fmt.Errorf("master device is (%s) not of type ipoib", lnk.Type())
+	}
+
+	ipoibLnk, ok := lnk.(*netlink.IPoIB)
+	if !ok {
+		return nil, fmt.Errorf("unexpected error, failed to convert to ipoib netlink interface")
+	}
+
+	// partition key is 15 bits
+	pkey := ipoibLnk.Pkey & 0x7fff
```

Comment:
reminder me why we need to do ipoibLnk.Pkey & 0x7fff?

Reply:
so full/partial pkey is determined in opensm?

Reply:
yes